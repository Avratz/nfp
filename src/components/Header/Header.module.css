.header {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
}
