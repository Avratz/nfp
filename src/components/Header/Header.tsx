'use client'
import dynamic from 'next/dynamic'

import { Container } from '@/ui/Container/Container'

import { Logo } from '@/components/Logo/Logo'
const Drawer = dynamic(
  async function () {
    return import('@/components/Drawer/Drawer').then((mod) => mod.Drawer)
  },
  {
    ssr: false,
    loading() {
      return null
    },
  },
)
import { Navbar } from '@/components/Navbar/Navbar'
import { ButtonMenu } from '@/components/ButtonMenu/ButtonMenu'

import { useOverlay } from '@/hooks/useOverlay'

import styles from './Header.module.css'

export function Header() {
  const { isOpen, setIsOpen } = useOverlay()
  return (
    <header className={styles.header}>
      <Container className={styles.headerContainer}>
        <Logo dark={isOpen} />
        <Navbar />
        <Drawer isOpen={isOpen} />
        <ButtonMenu isOpen={isOpen} setIsOpen={setIsOpen} />
      </Container>
    </header>
  )
}
