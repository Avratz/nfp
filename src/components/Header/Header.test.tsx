import { render } from '@testing-library/react'
import { Header } from './Header'

jest.mock('next/dynamic', () => () => {
  const DynamicComponent = () => null
  DynamicComponent.displayName = 'LoadableComponent'
  DynamicComponent.preload = jest.fn()
  return DynamicComponent
})

jest.mock('../../assets/images/paisanos-flag.svg', () => {
  const PaisanosFlag = () => null
  PaisanosFlag.displayName = 'PaisanosFlag'
  return PaisanosFlag
})

describe('Header', () => {
  it('renders Header unchanged', () => {
    let { container } = render(<Header />)
    expect(container).toMatchSnapshot()
  })
})
