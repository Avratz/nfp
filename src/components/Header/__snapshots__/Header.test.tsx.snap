// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header unchanged 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="headerContainer container containerPadding"
    >
      <a
        class="link logo"
        href="/"
      >
        <picture
          class="isotipo"
        />
        <h1
          class="heading 2xl inlineBlock heading"
        >
          NFPaisanos
        </h1>
      </a>
      <nav
        class="nav"
      >
        <hr
          class="divider vertical"
        />
        <a
          class="link link"
          href="/"
        >
          Discover
        </a>
        <a
          class="link link"
          href="/"
        >
          What we do
        </a>
        <a
          class="link button primary base button"
          href="/"
        >
          Connect Wallet
        </a>
      </nav>
      <nav
        class="nav"
        style="clip-path: inset(0% 0% 100% 0%);"
      >
        <div
          class="navInner"
        >
          <h3
            class="heading 2xl heading"
            style="opacity: 0; transform: translateY(20px) translateZ(0);"
          >
            Menu
          </h3>
          <a
            class="link link"
            href="/"
            style="opacity: 0; transform: translateY(20px) translateZ(0);"
          >
            Discover
          </a>
          <a
            class="link link"
            href="/"
            style="opacity: 0; transform: translateY(20px) translateZ(0);"
          >
            What we do
          </a>
          <div
            class="lineSeparation"
            style="opacity: 0; transform: translateY(20px) translateZ(0);"
          />
          <h3
            class="heading 2xl heading"
            style="opacity: 0; transform: translateY(20px) translateZ(0);"
          >
            Connect
          </h3>
          <a
            class="link button primaryDark base button"
            href="/"
            style="opacity: 0; transform: translateY(20px) translateZ(0);"
          >
            Connect Wallet
          </a>
        </div>
      </nav>
      <button
        class="button icon base menuButton"
        title="Menu"
      >
        <div
          class="bar top"
        />
        <div
          class="bar bottom"
        />
      </button>
    </div>
  </header>
</div>
`;
